src/hooks/useChangeSelect.js

```diff
@@ -0,0 +1,43 @@
+import {useReducer, useEffect} from 'react';
+import {getChangeKey} from '../utils';
+
+const useCollection = multiple => {
+    const update = (collection, {type, value}) => {
+        switch (type) {
+            case 'clear':
+                return collection.length ? [] : collection;
+            case 'toggle': {
+                if (multiple) {
+                    return collection.includes(value)
+                        ? collection.filter(item => item !== value)
+                        : collection.concat(value);
+                }
+
+                return [value];
+            }
+            default:
+                return collection;
+        }
+    };
+
+    const [collection, dispatch] = useReducer(update, []);
+    return {
+        collection,
+        clear() {
+            dispatch({type: 'clear'});
+        },
+        toggle(value) {
+            dispatch({value, type: 'toggle'});
+        },
+    };
+};
+
+export default (hunks, {multiple = false} = {}) => {
+    const {collection, clear, toggle} = useCollection(multiple);
+    useEffect(clear, [hunks]);
+
+    return [
```

Comment:
A tuple type is current suitable for our hooks, but I'm wondering if this can satisfy future possible changes, should we change the return value to an object to ensure more extensibility?

Reply:
Not a problem to me since:

```
const useNewApi = () => {
    const [...] = useChangeSelect(...)
    return {...}
}
```

We can decide it later before 2.0